From 22df74886c624e6f6cd9becc20cc8261d87b15a0 Mon Sep 17 00:00:00 2001
From: turbo-boo <59400127+turbo-boo@users.noreply.github.com>
Date: Sat, 11 Jul 2026 21:09:56 +0900
Subject: [PATCH 04/14] Revert "fingerprint: let UDFPS handlers forward
 required acquired events"

This reverts commit a7e4183eb0694ee0e5638cd933a80924aee97fd3.

diff --git a/aidl/fingerprint/Session.cpp b/aidl/fingerprint/Session.cpp
index ca66f00..b935731 100644
--- a/aidl/fingerprint/Session.cpp
+++ b/aidl/fingerprint/Session.cpp
@@ -380,9 +380,7 @@ void Session::notify(const fingerprint_msg_t* msg) {
             // don't process vendor messages further since frameworks try to disable
             // udfps display mode on vendor acquired messages but our sensors send a
             // vendor message during processing...
-            if (result != AcquiredInfo::VENDOR ||
-                (mUdfpsHandler && mUdfpsHandler->shouldForwardAcquired(
-                                          static_cast<int32_t>(result), vendorCode))) {
+            if (result != AcquiredInfo::VENDOR) {
                 mCb->onAcquired(result, vendorCode);
             }
         } break;
diff --git a/fingerprint/include/UdfpsHandler.h b/fingerprint/include/UdfpsHandler.h
index 3f6e007..78f2368 100644
--- a/fingerprint/include/UdfpsHandler.h
+++ b/fingerprint/include/UdfpsHandler.h
@@ -22,7 +22,6 @@ class UdfpsHandler {
     virtual void onFingerUp() {};
 
     virtual void onAcquired(int32_t result, int32_t vendorCode) {};
-    virtual bool shouldForwardAcquired(int32_t result, int32_t vendorCode) { return false; };
     virtual void onAuthenticationSucceeded() {};
     virtual void onAuthenticationFailed() {};
     virtual void cancel() {};
